website/content/api-docs/index.mdx

```diff
@@ -75,6 +75,29 @@ $ curl \
     http://127.0.0.1:8500/v1/kv/foo
 ```
 
+## Accepts URL-Encoded Resource Names
+
+Some Consul HTTP API endpoints require resource names in the URL. To pass a resource
+name containing characters that are invalid in a URL path segment, such as `/` or ` `,
+the resource name must be URL-encoded into the HTTP API call URL.
+
+Though any resource name *can* be used with the HTTP API if URL-encoded,
+we typically recommend using resource names that don't require URL-encoding anyways.
+Depending on the validation that Consul applies to a resource name, Consul may still
+reject a request if it considers the resource name invalid for that endpoint. And
+even if Consul considers the resource name valid, it may degrade other functionality,
+such as failed [DNS lookups](/docs/discovery/dns)
+for nodes or services with names containing invalid DNS characters.
+
+This HTTP API capability also allows the
+[CLI to accept arguments with URL-invalid characters](/commands#accepts-url-invalid-characters).
+
+### Exceptions: Characters that are Always Invalid
+
+The inclusion of certain characters in a URL resource name will cause a request
+to be rejected *even if the character is URL-encoded*, including:
+- Linefeed (`%0a`)
```

Comment:
@kisunji : could use your eyes for technical accuracy (since you're familiar with this code).

Reply:
Per Karl, I only gave this its own section for improved scannability

Reply:
I'm with @kisunji on this one -- especially because there appears to really only be one exception. Maybe change the section heading to something like "Invalid Characters" or something else so that the information would be scannable but topical enough to merit its own section?

Reply:
Changed to:

---

### Invalid Characters

The linefeed character (`%0a`) will cause a request to be rejected even if it is URL-encoded.